inc/olalevel_ticket.class.php

```diff
@@ -82,8 +82,9 @@ function getFromDBForTicket($ID, $olaType) {
          ],
          'LIMIT'        => 1
       ]);
-      if (count($iterator)) {
-         return $this->getFromIter($iterator);
+      if (count($iterator) == 1) {
+         $row = $iterator->next();
+         return $this->getFromDB($row['id']);
```

Comment:
Not OK since it will make one extra query on the database. I ad in mind getFromIter return one result only

Reply:
OK, I was wrong; change is OK, thank you!